trino/trino_test.go

```diff
@@ -1973,3 +1973,198 @@ func TestForwardAuthorizationHeader(t *testing.T) {
 
 	assert.NoError(t, db.Close())
 }
+
+func TestPagination(t *testing.T) {
+	var buf, buf2, buf3 *bytes.Buffer
+	var ts *httptest.Server
+	ts = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.URL.Path == "/v1/statement" {
+			if buf == nil {
+				buf = new(bytes.Buffer)
+
+				json.NewEncoder(buf).Encode(&stmtResponse{
+					ID:      "fake-query",
+					NextURI: ts.URL + "/v1/statement/20210817_140827_00000_arvdv/1",
+					Stats: stmtStats{
+						State: "QUEUED",
+					},
+				})
+			}
+			w.WriteHeader(http.StatusOK)
+			w.Write(buf.Bytes())
+			return
+		}
+
+		if r.URL.Path == "/v1/statement/20210817_140827_00000_arvdv/1" {
+			if buf2 == nil {
```

Comment:
Why are separate buffers needed here? Can we initialize them early, and remove this condition?